core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -459,6 +460,65 @@ public void testDataFilesTableSelection() throws IOException {
     Assert.assertEquals(expected, scan.schema().asStruct());
   }
 
+  @Test
+  public void testPartitionColumnNamedPartition() throws Exception {
+    TestTables.clearTables();
+    this.tableDir = temp.newFolder();
+    tableDir.delete();
+
+    Schema schema = new Schema(
+        required(1, "id", Types.IntegerType.get()),
+        // create a column named "partition", to recreate the problem in
+        // https://github.com/apache/iceberg/issues/3709
```

Comment:
Nit: This comment feels a little redundant given the test name. Unlike the Spark repo, we don't usually reference the issue when adding a test (though I think that's a good practice in general).

Maybe just `// See https://github.com/apache/iceberg/issues/3709` towards the top of the test if you feel it's necessary?

Reply:
I remove the comments. Thanks!